### DISCUSSION TITLE
Need clarification: Marten keeps regenerating code already generated when using TypeLoadMode.Auto

### DISCUSSION THREAD

**Discussion by natalie-o-perret** [JasperFx/marten#2908] [Q&A]
🙋‍♀️ Hey there,

Probably a misunderstanding, or maybe my expectations on me end aren't quite right, but this relates to this: 
https://mastodon.social/@oskardudycz@hachyderm.io/111758904050503603

I've checked the documentation a couple of times:
- https://martendb.io/configuration/prebuilding.html
- https://martendb.io/configuration/optimized_artifact_workflow.html

and my issue is that in the case of an "`IEvent` document" aka the `mt_events` table, even once the process has already created the generated code at runtime, it keeps recreating it (in the same process runtime).

Note: I created this ["repro"](https://github.com/natalie-o-perret/issues/tree/main/marten-unmanaged-memory-spikes)

**Comment by jeremydmiller** [comment]
@natalie-o-perret See the output about where it's putting the generated code. You're in "not nearly as commonly used" F# world where the project rules are a bit different, and Marten isn't putting the generated code in your actual project. W/ F# you'll need to have a separate C# project to hold the pre-generated code, and do some handholding for Marten to tell it where to put that generated code and where the "application assembly" is w/ the expected generated code.

Definitely an are where we could be better with docs and/or a sample project setup:(

**Reply by natalie-o-perret** [reply]
That being said, I find it odd now that retrospectively, you don't qualify this problem as a GitHub issue for Marten like I initially did. 
Not sure to follow why it is turned [the initial issue](https://github.com/JasperFx/marten/issues/2907) into this conversation 🤔. 

I mean this issue is literally preventing people from using the library in actual production setup if you're using a full F# implementation. And don't get me wrong, I'm not antagonizing here, I'm merely just trying to get the rationale behind this move 🤔.

**Reply by natalie-o-perret** [reply]
> about where it's putting the generated code. You're in "not nearly as commonly used" F# world where the project rules are a bit different, and Marten isn't putting the generated code in your actual project. W/ F# you'll need to have a separate C# project to hold the pre-generated code, and do some handholding

Considering what was done in [https://github.com/JasperFx/wolverine/issues/676](this), just wondering if things can be done these days without resorting to create an extra C# project?

**Comment by jeremydmiller** [comment]
There's a related issue in Wolverine: https://github.com/JasperFx/wolverine/issues/676
